cedar-policy-validator/src/coreschema.rs

```diff
@@ -389,11 +445,16 @@ pub mod request_validation_errors {
         pub fn action(&self) -> &ast::EntityUID {
             &self.action
         }
+
+        /// Resource types which actually are valid for that action
+        pub fn valid_resource_tys(&self) -> impl Iterator<Item = &ast::EntityType> {
+            self.valid_resource_tys.iter()
+        }
     }
 
     /// Context does not comply with the shape specified for the request action
     #[derive(Debug, Error, Diagnostic)]
-    #[error("context `{context}` is not valid for `{action}`")]
+    #[error("context `{}` is not valid for `{action}`", pretty_print(&.context))]
```

Comment:
this might be a good place to use the new thiserror syntax `#[error(fmt = path::to::myfmt)]` to avoid allocating a string here

Reply:
Performance on the error path is not critical, but seems reasonable to make a followup issue for this.  #1371